Update

gemansipator opened the pull request:
домашка 11 (4 задания)

gemansipator committed:
Task01 1. Создай список строк в методе main.
 2. Добавь в него 10 строчек с клавиатуры, но только добавлять не в конец списка, а в начало.
 3. Используя цикл выведи содержимое на экран, каждое значение с новой строки.

gemansipator committed:
Task02 1. Создай список строк.
        2. Добавь в него 5 строк с клавиатуры.
        3. Выполни 13 раз: удалить последнюю строку и вставить её в начало.
        4. Используя цикл выведи содержимое результирующего списка на экран, каждое значение с новой строки.

gemansipator committed:
Task03 1. Создай список строк.
        2. Добавь в него 10 строчек с клавиатуры.
        3. Узнай, какая строка в списке встретится раньше: самая короткая или самая длинная.
        Если таких строк несколько, то должны быть учтены самые первые из них.
        4. Выведи на экран строку из п.3. Должна быть выведена одна строка.

gemansipator committed:
Task04 1. Введи с клавиатуры 20 чисел, сохрани их в список и рассортируй по трём другим спискам:
       Число нацело делится на 3 (x%3==0), нацело делится на 2 (x%2==0) и все остальные.
       Числа, которые делятся на 3 и на 2 одновременно, например 6, попадают в оба списка.
       Порядок объявления списков очень важен.
       1.1 Создай метод printList
       2. Метод printList должен выводить на экран все элементы списка с новой строки.
       3. Используя метод printList выведи эти три списка на экран.
       Сначала тот, который для x%3, потом тот, который для x%2, потом последний.

RinatBeybutov commented on src/main/java/site/javadev/Lesson_11/Task03.java:
```diff
@@ -0,0 +1,45 @@
+        }
+
+        // Сравниваем индексы самой короткой и самой длинной строки
+        if (list.indexOf(minValueString) < list.indexOf(maxValueString)) {
```
посмотреть реализацию indexOf в ArrayList и на основе этой информации оптимизировать алгоритм программы

RinatBeybutov commented:
в самом начале задания написано провести сравнение для LinkedList и ArrayList. Поясню:
Нужно сравнить время работы программы для большого количества действий (например 100 тыс) для двух случаев: 
а. Вставка в конец списка
б. Вставка в начало списка

gemansipator committed:
Task03 убрал лишний indexOf и добавил переменные для хранения индекса

gemansipator committed:
сравнение для LinkedList и ArrayList.

gemansipator committed:
Убран повторный результат

RinatBeybutov commented on src/main/java/site/javadev/Lesson_11/ArrayListLinkedListSpeed.java:
```diff
@@ -0,0 +1,66 @@
+        long linkedListEndInsertTime = System.currentTimeMillis() - startTime;
+
+        // Тестирование вставки в начало списка
+        arrayList.clear();
```
В arrayList очищение работает так, что массив остается в прежнем размере, а все ссылки зануляются. Поэтому после очищения методом clear не будет выполняться расширение массива в данном алгоритме. Для чистого теста нужно создать новый arrayList. В итоге подтвердились гипотезы?)

gemansipator committed:
// Вместо очистки создаём новый ArrayList
        arrayList = new ArrayList<>();
        linkedList.clear(); чтобы массив не сохранял свой прежний размер.
